server/services/calculate-status-events-sums.js

```diff
@@ -0,0 +1,274 @@
+						var processorResult;
+
+						if (!stepResult.byProcessor[processorId]) stepResult.byProcessor[processorId] = {};
+
+						processorResult = stepResult.byProcessor[processorId];
+
+						oForEach(byProcessorData, function (statusData, status) {
+							var statusResult;
+
+							if (!processorResult[status]) processorResult[status] = { certificate: {} };
+
+							statusResult = processorResult[status];
+
+							// 3.2.1 [serviceName].processingStep[stepName]
+							//           .byProcessor[processorId][status].businessProcess
+							storeStatusData(statusResult, 'businessProcess', statusData);
+
+							statusData.forEach(function (bpId) {
+								var certificates = data.businessProcesses.get(bpId).certificates;
+
+								certificates.forEach(function (certificateName) {
+									// 3.2.2 [serviceName].processingStep[stepName]
+									//           .byProcessor[processorId][status].certificate[certificateName]
+									storeStatusData(statusResult.certificate, certificateName, statusData);
+								});
+							});
+						});
+					});
+				});
+			}
+		});
+	});
+
+	var eachStatusSetToSum = function (dataSet) {
+		Object.keys(dataSet).forEach(function (status) {
+			if (isSet(dataSet[status])) {
+				dataSet[status] = dataSet[status].size;
+			}
+		});
+	};
+
+	oForEach(resultMap, function (serviceResult) {
+		// 1 [serviceName].businessProcess[status]
+		eachStatusSetToSum(serviceResult.businessProcess);
+
+		// 2 [serviceName].certificate[certificateName][status]
+		oForEach(serviceResult.certificate, eachStatusSetToSum);
+
+		oForEach(serviceResult.processingStep, function (stepResult, stepName) {
+			stepResult.pending.businessProcess.forEach(function (bpId) {
+				var certificates = data.businessProcesses.get(bpId).certificates;
+
+				certificates.forEach(function (certificateName) {
+					// 3.1.2 [serviceName].processingStep[stepName].pending.certificate[certificateName]
+					storeStatusData(stepResult.pending.certificate, certificateName, [bpId]);
+				});
+			});
+
+			// 3.1.2 [serviceName].processingStep[stepName].pending.certificate[certificateName]
+			oForEach(stepResult.pending, eachStatusSetToSum);
+
+			// 3.1.1 [serviceName].processingStep[stepName].pending.businessProcess
+			stepResult.pending.businessProcess = stepResult.pending.businessProcess.size;
+
+			oForEach(stepResult.byProcessor, function (processorResult) {
+				// 3.2.1 [serviceName].processingStep[stepName]
+				//           .byProcessor[processorId][status].businessProcess
+				oForEach(processorResult, eachStatusSetToSum);
+
+				oForEach(processorResult, function (statusResult) {
+					// 3.2.2 [serviceName].processingStep[stepName]
+					//           .byProcessor[processorId][status].certificate[certificateName]
+					oForEach(statusResult, eachStatusSetToSum);
+				});
+			});
+		});
+	});
+
+	return resultMap;
+};
+
+module.exports = memoize(function (dateFrom, dateTo) {
```

Comment:
`dateFrom` and `dateTo` should be here db.Date instances, I see in code that they're assumed as ISO strings, it should not be like that.

Also memoized function should not be asynchronous, it complicates memoization handling unnecessary, it would be nicer as e.g:

```javascript

var getCalculate = getDatesMap()(function (datesMap) {
   return memoize(function (dateFrom, dateTo) { .. }) ;
});

module.exports = function (dateFrom, dateTo) {
  reutrn getCalculate()(function (calculate) { return calculate(dateFrom, dateTo); });
};
```